tests/utils/stl/test/config.py

```diff
@@ -0,0 +1,478 @@
+        stl_test_env['LIB'] = self.stl_lib_env_var
+        stl_test_env['PATH'] = self.stl_path_env_var
+
+        self.config.environment = stl_test_env
+
+    def configure_skip_list_location(self):
+        skip_list_path = self.get_lit_conf('skip_list_path', None)
+
+        if skip_list_path is not None:
+            self.skip_list_path = Path(skip_list_path)
+        else:
+            self.skip_list_path = Path(os.devnull)
+
+    def configure_skip_list_root(self):
+        skip_list_root = self.get_lit_conf('skip_list_root', None)
+
+        if skip_list_root is not None:
+            self.skip_list_root = Path(skip_list_root)
+        else:
+            self.skip_list_root = Path('')
+
+    def configure_excludes(self):
+        excludes = self.get_lit_conf('excludes', set())
+        additional_excludes = self.get_lit_conf('additional_excludes', '')
+
+        excludes.update(additional_excludes.split(','))
+
+        if self.skip_list_path is None:
+            self.configure_skip_list_location()
+
+        if self.skip_list_root is None:
+            self.configure_skip_list_root()
+
+        excludes.update(
+            map(lambda x: os.path.join(str(self.skip_list_root), Path(x)),
+                stl.test.file_parsing.parse_commented_file(
+                    self.skip_list_path)))
+
+        self.lit_config.excludes = excludes
+
+    def configure_expected_results_list_location(self):
+        expected_results_list_path = self.get_lit_conf(
+            'expected_results_list_path', None)
+
+        if expected_results_list_path is not None:
+            self.expected_results_list_path = Path(
+                expected_results_list_path)
+        else:
+            self.expected_results_list_path = Path(os.devnull)
+
+    def configure_expected_results_list_root(self):
+        expected_results_list_root = self.get_lit_conf(
+            'expected_results_list_root', None)
+
+        if expected_results_list_root is not None:
+            self.expected_results_list_root = Path(
+                expected_results_list_root)
+        else:
+            self.expected_results_list_root = Path('')
+
+    def configure_expected_results(self):
+        expected_results = self.get_lit_conf('expected_results', dict())
+
+        if self.expected_results_list_path is None:
+            self.configure_expected_results_list_location()
+
+        if self.expected_results_list_root is None:
+            self.configure_expected_results_list_root()
+
+        expected_results.update(
+            map(lambda x: (str(self.expected_results_list_root / x[0]), x[1]),
+                stl.test.file_parsing.parse_result_file(
+                    self.expected_results_list_path).items()))
+
+        self.lit_config.expected_results = expected_results
+        self.config.expected_results = \
+            getattr(self.config, 'expected_results', dict())
+
+    # TRANSITION: Have configuring the compiler have the same flow as
+    # everything else.

```

Comment:
What is "the same flow"? I don't know what that would look like. (Perhaps this is clear to a reader who knows Python.)

Reply:
Yeah that doesn't deserve a long-term `TRANSITION`. All the other "configure" functions look a certain way while this is the only odd one out is all I meant.